tests/import_export_tests.py

```diff
@@ -0,0 +1,386 @@
+"""Unit tests for Caravel"""
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+from __future__ import unicode_literals
+
+import json
+import pickle
+import unittest
+
+from caravel import db, models
+
+from .base_tests import CaravelTestCase
+
+
+class ImportExportTests(CaravelTestCase):
+    """Testing export import functionality for dashboards"""
+
+    examples_loaded = False
+
+    def __init__(self, *args, **kwargs):
+        super(ImportExportTests, self).__init__(*args, **kwargs)
+
+    def test_export_dashboards(self):
+        births_dash = db.session.query(models.Dashboard).filter_by(
+            slug='births').first()
+
+        # export 1 dashboard
+        export_dash_url_1 = (
+            '/dashboardmodelview/export_dashboards_form?id={}&action=go'
+            .format(births_dash.id)
+        )
+        resp_1 = self.client.get(export_dash_url_1)
+        exported_dashboards_1 = pickle.loads(resp_1.data)['dashboards']
+        self.assertEquals(1, len(exported_dashboards_1))
+        self.assertEquals('births', exported_dashboards_1[0].slug)
+        self.assertEquals('Births', exported_dashboards_1[0].dashboard_title)
+        self.assertEquals(
+            births_dash.id,
+            json.loads(exported_dashboards_1[0].json_metadata)['remote_id']
+        )
+
+        self.assertTrue(exported_dashboards_1[0].position_json)
+        self.assertEquals(9, len(exported_dashboards_1[0].slices))
+
+        exported_tables_1 = pickle.loads(resp_1.data)['datasources']
+        self.assertEquals(1, len(exported_tables_1))
+        self.assertEquals(
+            1, len([c for c in [t.columns for t in exported_tables_1]]))
+        self.assertEquals(
+            1, len([m for m in [t.metrics for t in exported_tables_1]]))
+
+        # export 2 dashboards
+        world_health_dash = db.session.query(models.Dashboard).filter_by(
+            slug='world_health').first()
+        export_dash_url_2 = (
+            '/dashboardmodelview/export_dashboards_form?id={}&id={}&action=go'
+            .format(births_dash.id, world_health_dash.id)
+        )
+        resp_2 = self.client.get(export_dash_url_2)
+        exported_dashboards_2 = pickle.loads(resp_2.data)['dashboards']
+        self.assertEquals(2, len(exported_dashboards_2))
+        self.assertEquals('births', exported_dashboards_2[0].slug)
+        self.assertEquals(9, len(exported_dashboards_2[0].slices))
```

Comment:
true, but I need a way to make sure that # of slices matches. I would prefer this over having a chance to miss a bug where duplicate slices are created